### PR TITLE
Refactor: Move dashboard's data processing to utils

### LINKED ISSUE #283: Huge discrepancy between CLI generated report and exported CSV
**Describe the bug**
Huge discrepancy between CLI generated report and exported CSV

**To Reproduce**
1. Generate report (websites that reproduced the issue on my end: https://www.mercadopago.com.ar, https://www.rakuten.co.jp) using [CLI](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/A.-PSAT's-How-To#build-and-run-psats-cli)
2. Open the report
3. Export the report via the export button on the report
4. Import the following reports on Google Sheet (or any spreadsheet) : `affected-cookies.csv`, `cookies.csv`, `report.csv`
5. Find the discrepancies in the # of cookies below

**Observed discrepancies**
1. Discrepancy in the total # of cookies: localhost report: **21** (1P cookies x 16 + 3P cookies x 5), exported report: **79** (1P x 16 + 3P x 63)
2. Discrepancy in the # of affected cookies: localhost report: **3**, exported report: **19**

**Expected behavior**
The # of cookies in the localhost report and exported report should be the same

**Screenshots**
1. localhost report > total cookies list screenshot
Screenshot 2023-11-20 at 11 42 00

2. exported report > total cookies list screenshot
Screenshot 2023-11-20 at 11 58 54

3.  localhost report > affected cookie list screenshot
Screenshot 2023-11-20 at 12 09 25

4. exported report > affected cookie list screenshot
Screenshot 2023-11-20 at 12 07 33

**Desktop (please complete the following information):**
 - OS: Macbook Pro Intel, Sonoma 14.1
 - PSAT version : [v0.3.2](https://github.com/GoogleChromeLabs/ps-analysis-tool/releases/tag/v0.3.2)

**Additional context**
- Also a minor bug that total # of cookies in the exported report > `report.csv` is wrong (it says 81 but expecting 79)
- The issue does not seem to reproduce for sites with smaller number of cookies.

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This PR extracts data processing procedures used for passing cookie and technology data in the presentational components of `cli-dashboard` package to the different utility functions. This makes automated unit testing in the `cli-dashboard` easier. 
In conjunction with that, this refactoring also makes the reuse of presentational components possible for the service (mentioned in issue #205).

## Screenshot/Screencast

---

Partially addresses #205
Partially fixes #283

### CODE REVIEW

File: packages/cli-dashboard/src/components/utils/extractReportData.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    Object.entries(_cData).forEach(([key, cookie]) => {
+      store[frame][key] = cookie;
+    });
+  });
+};
+
+export default extractReportData;
+export { extractCookies };
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/utils/extractReportData.ts:105]: @mayan-000 Would you mind making a separate file for `extractCookies` function.

File: packages/design-system/src/components/input/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+import { ClearIcon } from '../../icons';
+
+interface InputProps {
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  clearInput: () => void;
+}
+
+const Input = ({ value, onChange, clearInput }: InputProps) => {
```
Comment (ayushnirwal) [inline packages/design-system/src/components/input/index.tsx:33]: Input -> TextInput

File: packages/design-system/src/icons/clear-icon.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg width="7" height="7" viewBox="0 0 7 7" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ayushnirwal) [inline packages/design-system/src/icons/clear-icon.svg:1]: Try to use `fill="currentColor"` and give color by "color" style attribute. This way we can leverage Tailwinds design system bindings to color match.

**Review by ayushnirwal — CHANGES_REQUESTED** [review]
@mayan-000 Could you please also add unit tests for the utils that have been extracted out of the presentational components.

File: packages/design-system/src/icons/clear-icon.svg (hunk 3 above)
Comment (mayan-000) [reply packages/design-system/src/icons/clear-icon.svg:1]: Yes, this is being implemented at places as and when a requirement for reusability occurs with different colors for SVG.

File: packages/cli-dashboard/src/components/utils/extractCookies.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * Internal dependencies
+ */
+import type { CookieFrameStorageType, CookieJsonDataType } from '../../types';
+
+const extractCookies = (
+  cookieData: {
+    frameCookies: CookieFrameStorageType;
+  },
+  pageUrl: string,
+  isLandingPage = false
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/utils/extractCookies.ts:27]: NIT: I think this should be renamed as `shouldAddUrlToKey`. That way the context that landing page needs cookie keyed using page url will not be needed to understand this utility function.

File: packages/cli-dashboard/src/components/utils/tests/extractCookies.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          platform: 'Unknown',
+          category: 'Uncategorized',
+          GDPR: '',
+          description: '',
+          isFirstParty: true,
+          pageUrl,
+          frameUrl: 'https://edition.cnn.com',
+        },
+      },
+      'Unknown frame(s)': {},
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/utils/tests/extractCookies.ts:88]: Constant for this particular key has been added [here](https://github.com/GoogleChromeLabs/ps-analysis-tool/blob/eafeffb0f3bebe28e29a34e0681294a60998c3a5/packages/common/src/index.ts#L26).

File: packages/cli-dashboard/src/components/utils/tests/extractCookies.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
```
Comment (ayushnirwal) [inline packages/cli-dashboard/src/components/utils/tests/extractCookies.ts:20]: Correct me if I am wrong but I don't see any matchers added by matchers added by [jest-dom](https://github.com/testing-library/jest-dom#custom-matchers) used in this file.

**Review by ayushnirwal — CHANGES_REQUESTED** [review]

File: packages/design-system/src/components/input/index.tsx (hunk 2 above)
Comment (mayan-000) [reply packages/design-system/src/components/input/index.tsx]: From Input -> SearchInput

**Review by ayushnirwal — DISMISSED** [review]
LGTM

File: packages/common/src/cookies.types.ts [hunk 7]
```diff
@@ -60,7 +60,7 @@ export type CookieData = {
 
 export type CookieTableData = CookieData & {
   isCookieSet: boolean | null;
-  frameUrl?: string;
+  frameUrl?: string | string[];
```
Comment (ayushnirwal) [inline packages/common/src/cookies.types.ts:63]: Rename to frameUrls

**Review by ayushnirwal — APPROVED** [review]
LGTM.  Issue #286 looks resolved on my end. This PR can be merged after @kjnanda gives the green light.

**Pull request merged**